src/core/models/transaction.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  Model,
+  type InferAttributes,
+  type InferCreationAttributes,
+  type CreationOptional,
+  DataTypes
+} from 'sequelize'
+import { sequelize } from '../database/sequelize'
+import { Account } from './account'
+import { Tag } from './tag'
+import { type TransactionAttributes } from '../interfaces/transaction'
+
+export class Transaction
+  extends Model<
+  InferAttributes<Transaction>,
+  InferCreationAttributes<Transaction, { omit: 'id' }>
+  >
+  implements TransactionAttributes {
+  declare id: CreationOptional<number>
+  declare account_id: CreationOptional<number>
```

Comment:
Foreign keys should not be creationOptional since it's value is not set by default when we create a record in the table